public/app/features/explore/utils/links.ts

```diff
@@ -127,10 +128,34 @@ export const getFieldLinksForExplore = (options: {
         }
         return linkModel;
       } else {
+        let internalLinkSpecificVars: ScopedVars = {};
+        if (link.internal?.transformations) {
+          const fieldValue = field.values.get(rowIndex);
```

Comment:
Need to double check but I think we need to get the field from `transformation` here (it may be different than the "target field").

Reply:
 🤔 I think I would need to add another variable to the transformations model then. So I think what you're saying is the transformation should be applied to the transformation field (if set) or the target field (if not). That'll have to happen at a higher level to this, because this logic is only applied on the one field that's passed in - we'll want to be sure the transformation field gets it's own call. Does that sound right? 
 
 

Reply:
> because this logic is only applied on the one field that's passed in

This is the field where the link is attached to (for visualization), but we have access to all fields (via dataFrame object):

```typescript
export const getFieldLinksForExplore = (options: {
  field: Field;           // <--- where the link is displayed (used to get the config object)
  ...
  dataFrame?: DataFrame;  // <--- all the data (used to interpolate)
}):
```

It's actually currently possible to access values from other fields using `__data` variable in a link. For example you can write a query `"SELECT * FROM TABLE WHERE foo={$__data.fields.foo} AND bar={__data.fields.bar}"` such target query can be assigned to any target field.

Reply:
Oh right!  🤦 Thank you!